lib/membrane_rtc_engine/bitrate_estimator.ex

```diff
@@ -29,34 +29,37 @@ defmodule Membrane.RTC.Engine.BitrateEstimator do
     Map.update!(state, :sizes, &[entry | &1])
   end
 
-  @spec estimate(t()) :: {BitrateEstimation.t(), t()}
+  @spec estimate(t()) :: {:ok, BitrateEstimation.t(), t()} | {:error, :not_enough_data}
   def estimate(state) do
     start = state.last_estimation_time
     stop = get_timestamp()
     duration = Time.as_seconds(stop - start)
 
-    estimation =
-      case state.sizes do
-        [] ->
-          %BitrateEstimation{estimation: 0, error: 0}
-
-        sizes ->
-          %BitrateEstimation{
-            estimation:
-              sizes
-              |> Enum.sum()
-              |> Ratio.new(duration)
-              |> Ratio.to_float(),
-            error:
-              sizes
-              |> Statistics.stdev()
-              |> then(&(&1 * length(sizes)))
-              |> Ratio.new(duration)
-              |> Ratio.to_float()
-          }
-      end
+    cond do
+      duration == 0 ->
+        {:error, :not_enough_data}
+
+      state.sizes == [] ->
+        estimation = %BitrateEstimation{estimation: 0, error: 0}
+        {:ok, estimation, %{state | last_estimation_time: stop}}
```

Comment:
I'm pretty sure we no longer need that clause. I remember adding it because of an `ArithmeticError` (:see_no_evil:) that might have had something to do with dividing by zero. Anyhow, this will be handled nicely by the next clause.

The sum of sizes will be zero, as will the standard deviation of an empty set.

Reply:
std_dev of empty list is `nil`, that's why I put this clause

Reply:
Interesting. I would expect it to be 0